example_envs/bookinfo/leafnode.rs

```diff
@@ -32,25 +32,41 @@ impl SimElement for LeafNode {
                 let deq = self.core_node.dequeue(tick);
                 rpc = deq.unwrap();
             } else {
-                // no rpc in the queue, we only react so nothing to do
+                // no rpc in the queue, we only forward so nothing to do
                 continue;
             }
-            // "respond to every node that has sent a request"
+            // forward requests/responses from productpage or reviews
             if rpc.headers.contains_key("src") {
-                let target = rpc.headers["src"].to_string();
-                rpc.headers
-                    .insert("direction".to_string(), "response".to_string());
+                let dest = self.choose_destination(&rpc);
+                rpc.headers.insert("dest".to_string(), dest.clone());
                 rpc.headers
                     .insert("src".to_string(), self.core_node.id.to_string());
-                ret.push((rpc, target));
+                rpc.headers
+                    .insert("location".to_string(), "egress".to_string());
+                if self.core_node.plugin.is_some() {
```

Comment:
this code can be simplified, see https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html

Reply:
Good call.